products/dataproc/api.yaml

```diff
@@ -0,0 +1,186 @@
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'gracefulDecommissionTimeout'
+                required: true
+                description: |
+                  Timeout for YARN graceful decommissioning of Node Managers. Specifies the
+                  duration to wait for jobs to complete before forcefully removing workers
+                  (and potentially interrupting jobs). Only applicable to downscaling operations.
+
+                  Bounds: [0s, 1d].
+              - !ruby/object:Api::Type::Double
+                name: 'scaleUpFactor'
+                required: true
+                description: |
+                  Fraction of average pending memory in the last cooldown period for which to
+                  add workers. A scale-up factor of 1.0 will result in scaling up so that there
+                  is no pending memory remaining after the update (more aggressive scaling).
+                  A scale-up factor closer to 0 will result in a smaller magnitude of scaling up
+                  (less aggressive scaling).
+
+                  Bounds: [0.0, 1.0].
+              - !ruby/object:Api::Type::Double
+                name: 'scaleDownFactor'
+                required: true
+                description: |
+                  Fraction of average pending memory in the last cooldown period for which to
+                  remove workers. A scale-down factor of 1 will result in scaling down so that there
+                  is no available memory remaining after the update (more aggressive scaling).
+                  A scale-down factor of 0 disables removing workers, which can be beneficial for
+                  autoscaling a single job.
+
+                  Bounds: [0.0, 1.0].
+              - !ruby/object:Api::Type::Double
+                name: 'scaleUpMinWorkerFraction'
+                description: |
+                  Minimum scale-up threshold as a fraction of total cluster size before scaling
+                  occurs. For example, in a 20-worker cluster, a threshold of 0.1 means the autoscaler
+                  must recommend at least a 2-worker scale-up for the cluster to scale. A threshold of
+                  0 means the autoscaler will scale up on any recommended change.
+
+                  Bounds: [0.0, 1.0]. Default: 0.0.
+              - !ruby/object:Api::Type::Double
+                name: 'scaleDownMinWorkerFraction'
+                description: |
+                  Minimum scale-down threshold as a fraction of total cluster size before scaling occurs.
+                  For example, in a 20-worker cluster, a threshold of 0.1 means the autoscaler must
+                  recommend at least a 2 worker scale-down for the cluster to scale. A threshold of 0
+                  means the autoscaler will scale down on any recommended change.
+
+                  Bounds: [0.0, 1.0]. Default: 0.0.
+          - !ruby/object:Api::Type::String
```

Comment:
nit: This is mostly personal preference, but listing primitive fields before nested ones makes  definitions a little easier to read imo.